### ISSUE TITLE
[Bug] Full Node Overview Sync Reset to "0" and NOT Syncing

### ISSUE THREAD

**Issue by purupokharel** [Chia-Network/chia-blockchain#9952]
### What happened?

Chia GUI Full Node (fully synced node) reset to 0. Deleted the entire DB and synced back fully. Again, it reset to "0".

I am having this issue as this is a GUI full node on macOS Mojave 10.14.6 with 2 x 3.46zGHz 6-Core Intel Xeon and 96GB 1333 MHz DDR3 memory, that uses a 4TB NVMe SSD for plotting.

I thought the database might got corrupted, so I deleted the DB and synced back fully in a few days. For my surprize, immediately after it was fully synced, it got reset to ZERO again. Hence I deleted .chia folder from the home directory (it's a hidden folder). It took a while to fully sync the Node. But, a few minutes after the node was fully synced, surprisingly its status changed to **Not Synced** (in red), and then reset to ZERO just like before.

After the Sync status returned to 0/1,468,374 (no progress of syncing ever since), I think it must be a BUG. There is no firewall issue, or port issue at all. It is also nota server down issue. My other full nodes are functioning on the same server with no issue at all. 

I'm a bit curious here: does the Chia Mainnet has capacity to block a node, or blacklist it, or put it behind a firewall for whatever reason? I'm curious.

I have identical second Mac with the same configuration which is running smoothly in the same network. Same setup, same OS version, same memory specs, same processors specs. Surprisingly, one Mac fails to Sync, and other Mac doesn't have any issue at all. Any idea what is causing Chia Full Node sync reset to 0 after the Node was fully synced and working smoothly for almost a year?

One thing to note is -- if the minter never clicks on Full Node, he might never notice this bug as rest of the functions are working well such as you can continue plotting new plots.

Hope someone in the dev community realizes the bug and pushes the fix soon. Or someone in the community is smart enough to find a solution for this kind of Full Node Reset to ZERO issue? 

The following is the screenshot of what exactly looks like in my Full Node Overview:

![Screen Shot 2022-01-25 at 4 30 10 PM](https://user-images.githubusercontent.com/98131008/151062926-90dcd8d1-6e49-4f29-958d-e13995a2f420.png)

### Version

Version 1.2.11 | 64 bit

### What platform are you using?

macOS

### What ui mode are you using?

GUI

### Relevant log output

```shell
2022-01-25T16:03:20.445 full_node chia.rpc.rpc_server     : WARNING  Sending data failed. Exception Traceback (most recent call last):
  File "chia/rpc/rpc_server.py", line 69, in _state_changed
  File "aiohttp/client_ws.py", line 150, in send_str
  File "aiohttp/http_websocket.py", line 687, in send
  File "aiohttp/http_websocket.py", line 643, in _send_frame
  File "aiohttp/http_websocket.py", line 660, in _write
ConnectionResetError: Cannot write to closing transport
.
2022-01-25T16:03:20.453 full_node asyncio                 : ERROR    Task exception was never retrieved
future: <Task finished name='Task-87902' coro=<RpcServer._state_changed() done, defined at chia/rpc/rpc_server.py:48> exception=KeyError(b'\xd7\x80\xd2,z\x87\xc9\xe0\x1d\x98\xb4\x9a\t\x10\xf6p\x1c;\x95\x01WA1k?\xda\x04.]{\x81\xd2')>
Traceback (most recent call last):
  File "chia/rpc/rpc_server.py", line 51, in _state_changed
  File "chia/rpc/full_node_rpc_api.py", line 61, in _state_changed
  File "chia/rpc/full_node_rpc_api.py", line 102, in get_blockchain_state
  File "chia/consensus/blockchain.py", line 157, in get_peak
  File "chia/consensus/blockchain.py", line 674, in height_to_block_record
  File "chia/consensus/blockchain.py", line 670, in block_record
KeyError: b'\xd7\x80\xd2,z\x87\xc9\xe0\x1d\x98\xb4\x9a\t\x10\xf6p\x1c;\x95\x01WA1k?\xda\x04.]{\x81\xd2'
2022-01-25T16:03:23.017 full_node chia.rpc.rpc_server     : WARNING  Sending data failed. Exception Traceback (most recent call last):
  File "chia/rpc/rpc_server.py", line 69, in _state_changed
  File "aiohttp/client_ws.py", line 150, in send_str
  File "aiohttp/http_websocket.py", line 687, in send
  File "aiohttp/http_websocket.py", line 643, in _send_frame
  File "aiohttp/http_websocket.py", line 660, in _write
ConnectionResetError: Cannot write to closing transport
.
2022-01-25T16:03:23.018 full_node asyncio                 : ERROR    Task exception was never retrieved
future: <Task finished name='Task-87921' coro=<RpcServer._state_changed() done, defined at chia/rpc/rpc_server.py:48> exception=KeyError(b'\xd7\x80\xd2,z\x87\xc9\xe0\x1d\x98\xb4\x9a\t\x10\xf6p\x1c;\x95\x01WA1k?\xda\x04.]{\x81\xd2')>
Traceback (most recent call last):
  File "chia/rpc/rpc_server.py", line 51, in _state_changed
  File "chia/rpc/full_node_rpc_api.py", line 61, in _state_changed
  File "chia/rpc/full_node_rpc_api.py", line 102, in get_blockchain_state
  File "chia/consensus/blockchain.py", line 157, in get_peak
  File "chia/consensus/blockchain.py", line 674, in height_to_block_record
  File "chia/consensus/blockchain.py", line 670, in block_record
KeyError: b'\xd7\x80\xd2,z\x87\xc9\xe0\x1d\x98\xb4\x9a\t\x10\xf6p\x1c;\x95\x01WA1k?\xda\x04.]{\x81\xd2'
2022-01-25T16:03:23.021 full_node full_node_server        : ERROR    Exception <bytes32: d780d22c7a87c9e01d98b49a0910f6701c3b95015741316b3fda042e5d7b81d2>, exception Stack: Traceback (most recent call last):
  File "chia/server/server.py", line 292, in incoming_connection
  File "chia/server/server.py", line 335, in connection_added
  File "chia/full_node/full_node.py", line 631, in on_connect
  File "chia/consensus/blockchain.py", line 670, in block_record
KeyError: <bytes32: d780d22c7a87c9e01d98b49a0910f6701c3b95015741316b3fda042e5d7b81d2>

2022-01-25T16:03:23.023 full_node chia.rpc.rpc_server     : WARNING  Sending data failed. Exception Traceback (most recent call last):
  File "chia/rpc/rpc_server.py", line 69, in _state_changed
  File "aiohttp/client_ws.py", line 150, in send_str
  File "aiohttp/http_websocket.py", line 687, in send
  File "aiohttp/http_websocket.py", line 643, in _send_frame
  File "aiohttp/http_websocket.py", line 660, in _write
ConnectionResetError: Cannot write to closing transport
.
2022-01-25T16:03:23.024 full_node asyncio                 : ERROR    Task exception was never retrieved
future: <Task finished name='Task-87924' coro=<RpcServer._state_changed() done, defined at chia/rpc/rpc_server.py:48> exception=KeyError(b'\xd7\x80\xd2,z\x87\xc9\xe0\x1d\x98\xb4\x9a\t\x10\xf6p\x1c;\x95\x01WA1k?\xda\x04.]{\x81\xd2')>
Traceback (most recent call last):
  File "chia/rpc/rpc_server.py", line 51, in _state_changed
  File "chia/rpc/full_node_rpc_api.py", line 61, in _state_changed
  File "chia/rpc/full_node_rpc_api.py", line 102, in get_blockchain_state
  File "chia/consensus/blockchain.py", line 157, in get_peak
  File "chia/consensus/blockchain.py", line 674, in height_to_block_record
  File "chia/consensus/blockchain.py", line 670, in block_record
KeyError: b'\xd7\x80\xd2,z\x87\xc9\xe0\x1d\x98\xb4\x9a\t\x10\xf6p\x1c;\x95\x01WA1k?\xda\x04.]{\x81\xd2'
2022-01-25T16:03:23.912 full_node chia.rpc.rpc_server     : WARNING  Sending data failed. Exception Traceback (most recent call last):
  File "chia/rpc/rpc_server.py", line 69, in _state_changed
  File "aiohttp/client_ws.py", line 150, in send_str
  File "aiohttp/http_websocket.py", line 687, in send
  File "aiohttp/http_websocket.py", line 643, in _send_frame
  File "aiohttp/http_websocket.py", line 660, in _write
ConnectionResetError: Cannot write to closing transport
.
```

**Comment by emlowe** [comment]
This is another dup of #9677

How many times have you tried to resync and reproduce this problem?

**Comment by purupokharel** [comment]
Tried 3 times and reproduced the same results. Each time I shutdown the node, delete the **.chia** folder that contains the db and start the node, it begins to sync. Once the sync is completed, it resets to 0. Quite frustrating because full sync takes about 2 days time in my current Mac configurations. Not fun.

**Comment by emlowe** [comment]
Can you try it one more time - but this time can you make sure your log is in INFO mode and delete your log file before you start? If we can collect an entire log file from where you synd the chain and then this happens it would be helpful.

@mariano54

**Comment by JtheDBA** [comment]
I am having the exact same issue as of this morning. 8am EST.

GUI and CLI 1.2.11 on Windows 10 (Pro, in Hyper-V VM on Server 2019)

Have restarted the machine.

Should I delete the .chia/mainnet folder and restart in INFO mode as above?

**Comment by crazybanane** [comment]
I found an easy Solution for that.
-Close Chia 
-Download DB Browser (SQLite)
-Open DB Browser
-then open your DB > File> Open Database > Search for your blockchain db e.g. C:\Users\YOURUSERNAME\.chia\mainnet\db\blockchain_v1_mainnet.sqlite
-wait for loading
-go to TAB "Browse Data", you see now a Table with Numbers there is a Row/collumn with "is_peak" you have to click here until it shows under it a "1".
-now set it to "0" then click on the Row/column "height" until it shows the biggest number e.g. 144777 now set there "is_peak" to 1.
-Now go to >File >Write Database> File > Close Database.
-Start Chia Gui and wait a moment if it has loaded it must Show Syncing 144777/144899 after 3-5 minutes it starts Syncing.

**Comment by emlowe** [comment]
Currently, because we don't yet know why this is happening, we don't recommend anyone manually edit their blockchain DB.

**Comment by crazybanane** [comment]
Yeah for sure but its actual the only way to not resync the whole DB because losing Time 3-4Days. But it seems that this Problem is since December. I will Log too if there are new Sync Probs but its not nice to Resync every week........

**Comment by crazybanane** [comment]
Okay i made a shorted log (without keys) of the Problem how can i send it to an Admin?

**Comment by mariano54** [comment]
Yes can you send it to me on keybase, I'm sorgente711

**Comment by mariano54** [comment]
Can anyone that experienced the issue answer these questions:
1. What OS and OS version are you running?
2. How much RAM does your system have? did you ever get close to running out of RAM?
3. Are you running with default settings or did you change the config ?

**Comment by Uinden** [comment]
@mariano54 
I think I have same problem. Chia status says "Syncing 0/1,477,611"
Wallet status says "Not Synced (1,473,396)".
I run WIndows 10, 64gb of ram, never running out (swap (page file) disabled).
Default settings. Usual uptime ~2 months.
Will try DB fixing method https://github.com/Chia-Network/chia-blockchain/issues/9952#issuecomment-1022405441

**Comment by JtheDBA** [comment]
My system

1. Windows 10 Pro 21H2 (In Server 2019 Hyper-V VM)
2. 4GB, not that I know of (tuned down from 8GB to 4GB many months ago, which seemed optimal setting)
3. only changed debug = info and plot location in config

**Comment by crazybanane** [comment]
My System:
Win 10 Pro 21H1
48GB didnt run out of ram
Only change debug to INFO

**Comment by Listableble** [comment]
I have same problem
Ubuntu 20.04 LTS
128GB RAM
change debug to INFO

**Comment by ryu007za** [comment]
Also just experienced this issue, glad its not just me. 
Win10 Home
32GB RAM
Default config, only logging level changed

**Comment by ryu007za** [comment]
Also just experienced this issue, glad its not just me. 
Win10 Home
32GB RAM
Default config, only logging level changed

**Comment by booshila** [comment]
Same problem!
1. Windows 11 Pro
2. 16Gb RAM
3. Default

**Comment by kawww** [comment]
Same

1.Ubuntu 20.04.3 LTS
2.2 GB RAM
3.Default

**Comment by mariano54** [comment]
Please see my response in #9677 . The latest  @beta version can be installed from github manually or in chia.net downloads, but be careful as it's beta.

**Comment by kathekon** [comment]
Same problem!  

1. Ubuntu 20.04.3 LTS
2. 16 gig
3. Default with "Info" logging level

**Comment by MrSavDre** [comment]
> I found an easy Solution for that. -Close Chia -Download DB Browser (SQLite) -Open DB Browser -then open your DB > File> Open Database > Search for your blockchain db e.g. C:\Users\YOURUSERNAME.chia\mainnet\db\blockchain_v1_mainnet.sqlite -wait for loading -go to TAB "Browse Data", you see now a Table with Numbers there is a Row/collumn with "is_peak" you have to click here until it shows under it a "1". -now set it to "0" then click on the Row/column "height" until it shows the biggest number e.g. 144777 now set there "is_peak" to 1. -Now go to >File >Write Database> File > Close Database. -Start Chia Gui and wait a moment if it has loaded it must Show Syncing 144777/144899 after 3-5 minutes it starts Syncing.

Thank you very much!

**Comment by pmarson** [comment]
quick way to find the max height in DB Browser - run this query in the Execute SQL tab

SELECT MAX(HEIGHT) FROM block_records

The max height will be shown in the results window, then you can easily find that record using the filter in the height column (just enter the number you are looking for)

**Comment by ericthemantis** [comment]
Downloaded the beta and wallet already sync'd. Will likely take a couple days for DB to sync, but will report back.

**Comment by bioshazard** [comment]
After a `stop` and `chia start farmer` today, I found my DB was reset to zero. Took a few minutes but it appears to be syncing now.

**Comment by bioshazard** [comment]
From CLI, seeing 0 in `chia show -s` and `Feb 28 16:18:51 full_node Start syncing from fork point at 0 up to 1630339` in logs.

**Comment by emlowe** [comment]
This issue should be fixed in 1.3 release - however, if your DB is currently in this state you will need to resync.

**Comment by bioshazard** [comment]
FWIW I am back in action after restoring a 40G db backup. Took a couple days to catch up and has been fine on the same version.
